File: gateOne/day 7/javascript/atmapp.js
const prompt = require("prompt-sync")();

let profile = [];
let phoneNo = [];
let pin = [];
let accountBalance = [];





function header(){

		console.log("===============================");

		console.log("WELCOME TO BANKE BANK");

		console.log("===============================");

}

function createAccountName(){
	const prompt = require("prompt-sync")();
	let userNames = prompt("Enter first name and last name: ");
	return userNames;
}

function phoneNumber(){
	const prompt = require("prompt-sync")();
	let userPhoneNumber = prompt("Enter phone number: ");
	return userPhoneNumber;
}

function createPin(){
	const prompt = require("prompt-sync")();
	let userPin = prompt("Enter pin: ");
	let pinLength = userPin.length;
	if(pinLength < 4 || pinLength > 4){
		console.log("Wrong pin!!! logging out");
	}else{
	return userPin;
	}
}

function createAccount(){
	let createNewAccount;
	let counter = 0;
	let balance = 0;
	do{
		profile.push(createAccountName());
		phoneNo.push(phoneNumber());
		pin.push(createPin());
		accountBalance.push(balance);
		console.log(">>>>>>>>Account created successfully<<<<<<<<<<<<");
		createNewAccount = prompt("Do you want to create a new Account? ");
		counter++;
	}while(createNewAccount == "yes");
	let desicion = prompt("Do you want to continue? ");
	if(desicion == "yes"){
		menu();
	}

}


function deposit(){
	let name2 = prompt("Enter account name: ");
	let userDepositAmount = Number(prompt("Enter deposit amount: "));
	for(let count2 = 0; count2 < accountBalance.length; count2++){
		if(name2 == profile[count2]){
			accountBalance[count2] = accountBalance[count2] + userDepositAmount;
			console.log("You just deposited " + userDepositAmount);
			console.log("Your account balance is " + accountBalance[count2]);
		}
	}
	let desicion2 = prompt("Do you want to continue? ");
	if(desicion2 == "yes"){
		menu();
	}
}


function withdrawal(){
	let name3 = prompt("Enter account name: ");
	let userWithdrawAmount = Number(prompt("Enter withdraw amount: "));
	for(let count3 = 0; count3 < accountBalance.length; count3++){
		if(name3 == profile[count3]){
			accountBalance[count3] = accountBalance[count3] - userWithdrawAmount;
			console.log("You have been debited " + userWithdrawAmount);
			console.log("Your account balance is " + accountBalance[count3]);
		}
	}
	let desicion3 = prompt("Do you want to continue? ");
	if(desicion3 == "yes"){
		menu();
	}
}

function transfer(){
	let enterRecipentName = prompt("Enter Recipent account name: ");
	let name4 = prompt("Enter your account name: ");
	let inputPin = prompt("Enter your pin: ");
	let userTransferAmount = Number(prompt("Enter transfer amount: "));
	for(let index4 = 0; index4 < accountBalance.length; index4++){
		if(inputPin == pin[index4]){
			if(userTransferAmount <= accountBalance[index4]){
				accountBalance[index4] -= userTransferAmount;
				for(let count4 = 0; count4 < accountBalance.length; count4++){
					if(enterRecipentName == profile[count4]){
					accountBalance[count4] = accountBalance[count4] + userTransferAmount;
					console.log("Dear " + enterRecipentName + " You just received " + userTransferAmount + " from " + name4);
					}
				}
			}else{
				console.log("Transaction failed!!! Account balance is too low")
			}

		}else{
			console.log("Account doesn't exist");
		}
		
	let desicion4 = prompt("Do you want to continue? ");
	if(desicion4 == "yes"){
		menu();
	}
	}
}


function changePin(){
let oldPin = prompt("Enter old pin: ");
for(let index5 = 0; index5 < accountBalance.length; index5++){
	if(oldPin == pin[index5]){
		let newPin = prompt("Enter new pin: ");
		pin[index5] = newPin;
		console.log("Your new pin is " + pin[index5]);
		break;
	}
}
	let desicion5 = prompt("Do you want to continue? ");
	if(desicion5 == "yes"){
		menu();
	}
}

function checkBalance(){
	let name6 = prompt("Enter account name: ");
	for(let index6 = 0; index6 < accountBalance.length; index6++){
		if(name6 == profile[index6]){
			console.log("Your account balance is " + accountBalance[index6]);
			break;
		}else
	}
	let desicion6 = prompt("Do you want to continue? ");
	if(desicion6 == "yes"){
		menu();
	}
}

function menu(){
	console.log("");
console.log("CHOOSE YOUR PREFERED OPTION", "\n", "1. Create Account", "\n", "2. Deposit", "\n", "3. Withdraw", "\n", "4. Transfer", "\n", "5. Change pin", "\n", "6. Check Balance", "\n", "7. Back to Main Menu");
console.log("");
let select = Number(prompt("Enter number: "));
	switch(select){
		case 1:
			createAccount();
			break;
		case 2:
			deposit();
			break;
		case 3:
			withdrawal();
			break;
		case 4:
			transfer();
			break;
		case 5:
			changePin();
			break;
		case 6:
			checkBalance();
			break;
		case 7:
			menu();
			
	}
}


header()
console.log("");
console.log("CHOOSE YOUR PREFERED OPTION", "\n", "1. Create Account", "\n", "2. Deposit", "\n", "3. Withdraw", "\n", "4. Transfer", "\n", "5. Change pin", "\n", "6. Check Balance", "\n", "7. Back to Main Menu");
console.log("");
let select = Number(prompt("Enter number: "));
	switch(select){
		case 1:
			createAccount();
			break;
		case 2:
			deposit();
			break;
		case 3:
			withdrawal();
			break;
		case 4:
			transfer();
			break;
		case 5:
			changePin();
			break;
		case 6:
			checkBalance();
			break;
		case 7:
			menu();
			
	}

File: gateOne/day 7/python/atmapp.py
profile_name = []
phone_no = []
pin = []
account_balance = []

print("=====================================")
print("WELCOME TO BANKE BANK")
print("=====================================")


def create_account():
	user_name = str(input("Enter first name and last name: "))
	profile_name.append(user_name);
	user_phone_number = str(input("Enter phone number: "))
	phone_no.append(user_phone_number)
	user_pin = str(input("Enter pin: "))
	pin.append(user_pin)
	account_balance.append(0)
	print(">>>>>>>>>> Account created successfully <<<<<<<<<<<<")
	decision = str(input("Do you want to create another account? "))
	while decision.lower() == "yes":
		user_name = str(input("Enter first name and last name: "))
		profile_name.append(user_name);
		user_phone_number = str(input("Enter phone number: "))
		phone_no.append(user_phone_number)
		user_pin = str(input("Enter pin: "))
		pin.append(user_pin)
		account_balance.append(0)
		print(">>>>>>>>>> Account created successfully <<<<<<<<<<<<")
		decision = str(input("Do you want to create another account? "))
		if decision.lower() != "yes":
			break 
	choice = str(input("Do you want to continue? "))
	if choice.lower() == "yes":
		menu()

def deposit():
	user_name = str(input("Enter account name: "))
	user_deposit_amt = int(input("Enter deposit amount: "))
	for index in range(len(profile_name)):
		if user_name == profile_name[index]:
			account_balance[index] = account_balance[index] + user_deposit_amt
			print("You just deposited", user_deposit_amt, "naira")
			print("Your account balance is ", account_balance[index])
	choice = str(input("Do you want to continue? "))
	if choice.lower() == "yes":
		menu()

def withdrawal():
	user_name = str(input("Enter account name: "))
	user_withdraw_amt = int(input("Enter withdrawal amount: "))
	for index in range(len(profile_name)):
		if user_name == profile_name[index]:
			if user_withdraw_amt <= account_balance[index]:
				account_balance[index] = account_balance[index] - user_withdraw_amt
				print("You have been debited ", user_withdraw_amt, "naira")
				print("Your account balance is", account_balance[index], "naira")
				break
			elif user_withdraw_amt > account_balance[index]:
				print("Your account balance is too low")
				break
	choice = str(input("Do you want to continue? "))
	if choice.lower() == "yes":
		menu()


def transfer():
	count = 0
	counter = 0
	recipent_name = str(input("Enter recipent account name: "))
	user_name = str(input("Enter your account name: "))
	user_pin = str(input("Enter pin: "))
	user_transfer_amt = int(input("Enter amount: "))
	for index in profile_name:
		if user_name == index and user_transfer_amt <= account_balance[count]:
			account_balance[count] = account_balance[count] - user_transfer_amt
			for element in profile_name:
				if recipent_name == element:
					account_balance[counter] = account_balance[counter] + user_transfer_amt
					print("Dear", recipent_name, "you just received", user_transfer_amt, "naira from", user_name)
				counter += 1
		elif user_name != index and user_transfer_amt > account_balance[count]:
			print("Transaction failed!! Your balance is too low")
			break
		count += 1
	choice = str(input("Do you want to continue? "))
	if choice.lower() == "yes":
		menu()


def change_pin():
	count1 = 0
	old_pin = str(input("Enter old pin: "))
	new_pin = str(input("Enter new pin: "))
	for element in pin:
		if old_pin == element:
			pin[count1] = new_pin
			print("your new pin is", pin[count1])
			break
		count1 += 1
	choice = str(input("Do you want to continue? "))
	if choice.lower() == "yes":
		menu()

			
def check_balance():
	count2 = 0
	user_name = str(input("Enter your account name: "))
	collect_pin = str(input("Enter your pin: "))
	for element in profile_name:
		if user_name == element:
			print("Your account balance is ",account_balance[count2])
			break
		count2 += 1
	choice = str(input("Do you want to continue? "))
	if choice.lower() == "yes":
		menu()

def menu():
	print("CHOOSE PERFERRED OPTION\n"
	"1 -> Create account\n"
	"2 -> Deposit\n"
	"3 -> Withdraw\n"
	"4 -> Transfer\n"
	"5 -> Change pin\n"
	"6 -> Check balance\n"
	"7 -> Back to menu\n")

	select = int(input("Enter preferred number: "))	
	match select:
		case 1: create_account()
		case 2: deposit()
		case 3: withdrawal()
		case 4: transfer()
		case 5: change_pin()
		case 6: check_balance()
		case 7: menu()
			
	
					
				
print("CHOOSE PERFERRED OPTION\n"
	"1 -> Create account\n"
	"2 -> Deposit\n"
	"3 -> Withdraw\n"
	"4 -> Transfer\n"
	"5 -> Change pin\n"
	"6 -> Check balance\n"
	"7 -> Back to menu\n")

select = int(input("Enter preferred number: "))	
match select:
	case 1: create_account()
	case 2: deposit()
	case 3: withdrawal()
	case 4: transfer()
	case 5: change_pin()
	case 6: check_balance()
	case 7: menu()
				
			
		
